pkg/modulereader/resreader_test.go

```diff
@@ -98,6 +102,64 @@ func (s *MySuite) TestFactory(c *C) {
 	c.Assert(reflect.TypeOf(tfReader), Equals, reflect.TypeOf(TFReader{}))
 }
 
+func (s *MySuite) TestGetModuleInfo_Embedded(c *C) {
+	ModuleFS = testModuleFS
+
+	// Success
+	moduleInfo, err := GetModuleInfo("modules/test_role/test_module", tfKindString)
+	c.Assert(err, IsNil)
+	c.Assert(moduleInfo.Inputs[0].Name, Equals, "test_variable")
+	c.Assert(moduleInfo.Outputs[0].Name, Equals, "test_output")
+
+	// Invalid: No embedded modules
+	badEmbeddedMod := "modules/does/not/exist"
+	moduleInfo, err = GetModuleInfo(badEmbeddedMod, tfKindString)
+	expectedErr := "failed to get info using tfconfig for terraform module at .*"
+	c.Assert(err, ErrorMatches, expectedErr)
+
+	// Invalid: Unsupported Module Source
+	badSource := "gcs::https://www.googleapis.com/storage/v1/GoogleCloudPlatform/hpc-toolkit/modules"
+	moduleInfo, err = GetModuleInfo(badSource, tfKindString)
+	expectedErr = "Source is not valid: .*"
+	c.Assert(err, ErrorMatches, expectedErr)
+}
+
+func (s *MySuite) TestGetModuleInfo_Git(c *C) {
+
+	// Invalid git repository - path does not exists
+	badGitRepo := "github.com:not/exist.git"
```

Comment:
Do these tests for GitHub and https URIs introduce an implied expectation of network connectivity to run unit tests? (i.e. testing for failure implies we expect a good value to succeed)

I believe (without full review) that we do not otherwise actually require GCP credentials or network connections.

Reply:
It looks like there is a similar failure in the sourcereader tests when network is cut off, but this test passes (test logs below).

These tests as are were copied over as is from sourcereader. Given that we have other tests that require network connection, I'd prefer to add a follow-up task to handle it. It would also make sense to run PR-Validation tests without the network once that's fixed. 

Let me know if it's enough of a concern to update this now, otherwise I'll consider this comment resolved once the task is created.

```
$ unshare -r -n go test hpc-toolkit/pkg/...
ok      hpc-toolkit/pkg/config  (cached)
ok      hpc-toolkit/pkg/deploymentio    (cached)
ok      hpc-toolkit/pkg/modulereader    (cached)
ok      hpc-toolkit/pkg/modulewriter    (cached)
?       hpc-toolkit/pkg/validators      [no test files]

----------------------------------------------------------------------
FAIL: git_test.go:25: MySuite.TestCopyGitModules

git_test.go:35:
    c.Assert(err, IsNil)
... value *errors.errorString = &errors.errorString{s:"error downloading 'https://github.com/terraform-google-modules/terraform-google-project-factory.git': /usr/bin/git exited with 128: Cloning into '/tmp/getter1748151968/temp'...\nssh: Could not resolve hostname github.com.corp.google.com.: Temporary failure in name resolution\r\nssh: Could not resolve hostname github.com.prod.google.com.: Temporary failure in name resolution\r\nssh: Could not resolve hostname github.com.prodz.google.com.: Temporary failure in name resolution\r\nssh: Could not resolve hostname github.com.net.google.com.: Temporary failure in name resolution\r\nssh: Could not resolve hostname github.com.google.com.: Temporary failure in name resolution\r\nssh: Could not resolve hostname github.com.in.goog.: Temporary failure in name resolution\r\nssh: Could not resolve hostname github.com: Temporary failure in name resolution\r\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists.\n"} ("error downloading 'https://github.com/terraform-google-modules/terraform-google-project-factory.git': /usr/bin/git exited with 128: Cloning into '/tmp/getter1748151968/temp'...\nssh: Could not resolve hostname github.com.corp.google.com.: Temporary failure in name resolution\r\nssh: Could not resolve hostname github.com.prod.google.com.: Temporary failure in name resolution\r\nssh: Could not resolve hostname github.com.prodz.google.com.: Temporary failure in name resolution\r\nssh: Could not resolve hostname github.com.net.google.com.: Temporary failure in name resolution\r\nssh: Could not resolve hostname github.com.google.com.: Temporary failure in name resolution\r\nssh: Could not resolve hostname github.com.in.goog.: Temporary failure in name resolution\r\nssh: Could not resolve hostname github.com: Temporary failure in name resolution\r\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists.\n")

OOPS: 10 passed, 1 FAILED
--- FAIL: Test (0.10s)
FAIL
FAIL    hpc-toolkit/pkg/sourcereader    0.219s
FAIL
```